### PR TITLE
Added documentation for new features of RMT

### PR DESCRIPTION
**Pull request by tmuntaner**
### Description

Added documentation for new RMT features:

* rmt-cli repos clean
* rmt-cli systems remove
* "rmt-cli mirror" subcommands
* web server performance settings

Please hold on merging this pull request until the features are released.

### To which product versions do the changes apply?

The first column is to be filled by the requester, the second column is to be filled by the doc team after the changes have been backported to the maintenance branches.

| Code 15     | Backport Done
| ----------  | ---------- 
| [x] 15 SP3  | (`master` only)   
| [x] 15 SP2  | [x] 
| [x] 15 SP1  | [ ] 
| [x] 15 SP0  | [ ]

### CODE REVIEW

**Comment by jmammen** [comment]
LGTM

**Comment by taroth21** [comment]
@tmuntaner : Thank you very much for your PR! We have recently changed our policy and are ready now to also  merge PRs that are exclusively for SP3. Please let us know whenever this one is content-wise finished (or if it already is now?), then we will do a quick review and merge it.

**Comment by tmuntaner** [comment]
@taroth21 Thanks, we're going to make a submit request to all SLES 15 requests today.

All SLES 15 versions will have the same version of RMT, but as we don't introduce any breaking changes (just new features), we may just release this document for the latest SLES. If that is SLES 15 SP3, then we would be happy to release it there.

**Comment by taroth21** [comment]
@tmuntaner : Sorry, I'm not sure I got you correctly... Do the changes in this PR apply to all Code 15 SLE versions? If yes, I would merge them into master first (for the upcoming SLE 15 SP3 release, following our 'master first' policy). I would also take care of backporting them to the respective maintenance branches for the SP2 -> SP0 (so you would not need to open PRs for the older Code 15 versions).

**Comment by tmuntaner** [comment]
@taroth21 yes, this change would apply to all SLE 15 versions. Technically, the entire book should be valid for all SLE 15 versions.

Also, the backporting would be great, but I have a question: is the process to apply the changes right away or wait for the release with the changes to be published?

Lastly, should I merge or you? I'm not sure of the procedure here.

**Comment by taroth21** [comment]
> @taroth21 yes, this change would apply to all SLE 15 versions. Technically, the entire book should be valid for all SLE 15 versions.

Got it now, thanks.
 
> Also, the backporting would be great, but I have a question: is the process to apply the changes right away or wait for the release with the changes to be published?

Yes, syncing doc updates with software releases makes sense (and sorry, I must have overlooked your sentence from the PR description on Friday: 'Please hold on merging this pull request until the features are released.') ;)

In that case, give us a shout after the features have been submitted for SP3, then we can merge the PR to `master`. 

And please give us another shout after the software changes have passed the maintenance process for SP2, SP1 and 15 GA. (For Code 15, we publish most of the documentation online only, so it's no problem to update documentation.suse.com at any time after the software update has been published).

> Lastly, should I merge or you? I'm not sure of the procedure here.

One of us (doc team) has to merge. 

Thank you!

File: xml/rmt_config_files.xml [hunk 1]
```diff
@@ -164,6 +164,47 @@
...
+   <title>Web Server Settings</title>
+   <para>
+    The <literal>web_server</literal> section lets you tune the performance of your RMT server.
+   </para>
+   <variablelist>
+    <varlistentry>
+     <term><literal>min_threads</literal></term>
+     <listitem>
+      <para>
+       Specifies the minimum threads count a RMT server worker should spawn.
```
Comment (tbazant) [inline xml/rmt_config_files.xml:177]:
```suggestion
       Specifies the minimum number of threads that an RMT server worker should spawn.
```

File: xml/rmt_config_files.xml [hunk 2]
```diff
@@ -164,6 +164,47 @@
...
+      <para>
+        Acceptable values: Integer greater than or equal to 1.
+      </para>
+     </listitem>
+    </varlistentry>
+    <varlistentry>
+     <term><literal>max_threads</literal></term>
+     <listitem>
+      <para>
+       Specifies the maximum threads count a RMT server worker should spawn.
```
Comment (tbazant) [inline xml/rmt_config_files.xml:188]:
```suggestion
       Specifies the maximum number of threads that an RMT server worker should spawn.
```

File: xml/rmt_config_files.xml [hunk 3]
```diff
@@ -164,6 +164,47 @@
...
+      <para>
+       Acceptable values: Integer greater than or equal to 1.
+      </para>
+     </listitem>
+    </varlistentry>
+    <varlistentry>
+     <term><literal>workers</literal></term>
+     <listitem>
+      <para>
+       Specifies the number of web workers for RMT.
```
Comment (tbazant) [inline xml/rmt_config_files.xml:199]:
```suggestion
       Specifies the number of Web workers for RMT.
```

File: xml/rmt_mirroring.xml [hunk 4]
```diff
@@ -72,7 +72,7 @@
    </step>
    <step>
     <para>
-     If you are member of multiple organizations, chose the organization you
+     If you are member of multiple organizations, choose the organization you
```
Comment (tbazant) [inline xml/rmt_mirroring.xml:75]:
```suggestion
     If you are a member of multiple organizations, select the organization you
```

File: xml/rmt_mirroring.xml [hunk 5]
```diff
@@ -221,36 +259,76 @@ Jun 22 04:22:34 d31 systemd[1]: Started RMT Mirror timer.</screen>
...
  <sect1 xml:id="sec-rmt-mirroring-delete">
   <title>Deleting Mirrored Data</title>
   <para>
-   After you disable mirroring of a repository or product as described in <xref linkend="sec-rmt-mirroring-enable-disable"/>, the mirrored data still remains on your local hard disk. This includes the mirrored RPM packages.
-   To remove the data, manually remove the corresponding directory:
+   After you disable the mirroring of a repository or product as described in <xref linkend="sec-rmt-mirroring-enable-disable"/>, the mirrored data still remains on your local hard disk. This includes the mirrored RPM packages.
   </para>
-  <screen>&prompt.sudo;<command>rm -r /usr/share/rmt/public/repo/SUSE/Products/<replaceable>PRODUCT</replaceable>/<replaceable>VERSION</replaceable>/<replaceable>ARCHITECTURE</replaceable>/</command></screen>
+  <para>
+   To delete a disabled repository's data, use the command <command>rmt-cli repos clean</command>. With this command, RMT
```
Comment (tbazant) [inline xml/rmt_mirroring.xml:303]:
```suggestion
   To delete disabled repository's data, use the command <command>rmt-cli repos clean</command>. With this command, RMT
```

File: xml/rmt_rmt-cli.xml [hunk 6]
```diff
@@ -258,6 +272,60 @@
...
+     <para>
+      This command lists registered systems.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>rmt-cli systems scc-sync</command></term>
+    <listitem>
+     <para>
+      This command forward registered systems data to SCC.
```
Comment (tbazant) [inline xml/rmt_rmt-cli.xml:317]:
```suggestion
      This command forwards registered systems data to SCC.
```

File: xml/rmt_rmt-cli.xml [hunk 7]
```diff
@@ -258,6 +272,60 @@
...
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>rmt-cli systems remove [TARGET]</command></term>
+    <listitem>
+     <para>
+      This command removes a system from RMT by its login.
```
Comment (tbazant) [inline xml/rmt_rmt-cli.xml:325]: i dont understand what "by its login" means. Can you maybe share more words that would explain it?

**Review by tbazant — CHANGES_REQUESTED** [review]
few nits + question

File: xml/rmt_mirroring.xml (hunk 5 above)
Comment (lproven) [reply xml/rmt_mirroring.xml:303]: No, I would suggest reverting this change.

**Review by lproven — CHANGES_REQUESTED** [review]

File: xml/rmt_mirroring.xml (hunk 5 above)
Comment (tbazant) [inline xml/rmt_mirroring.xml:303]:
```suggestion
   To delete disabled repository's data, use the command <command>rmt-cli repos clean</command>. With this command, RMT
```

File: xml/rmt_rmt-cli.xml (hunk 7 above)
Comment (tmuntaner) [reply xml/rmt_rmt-cli.xml:325]: HI @tbazant, `login` is a column in the systems list:

```
bin/rmt-cli systems list
```

```                                                                                                                                                                                                    Tue 3 10:03AM
+--------------------------------------+------------+-------------------------+-------------------------+------------------------------------------+
| Login                                | Hostname   | Registration time       | Last seen               | Products                                 |
+--------------------------------------+------------+-------------------------+-------------------------+------------------------------------------+
| SCC_bc0d2a200d884bddaf0f51e31165193b | linux-h8ph | 2020-09-08 15:42:04 UTC | 2020-09-08 15:42:29 UTC | SLES/15/x86_64                           |
|                                      |            |                         |                         | sle-module-basesystem/15/x86_64          |
|                                      |            |                         |                         | sle-module-server-applications/15/x86_64 |
| SCC_476bbc7fa9d54c0b8ae655c0f98ebbc5 | linux-h8ph | 2020-09-08 15:40:55 UTC | 2020-09-08 15:41:22 UTC | SLES/15/x86_64                           |
|                                      |            |                         |                         | sle-module-basesystem/15/x86_64          |
|                                      |            |                         |                         | sle-module-server-applications/15/x86_64 |
| SCC_ee59a04ad6fe4b05a056b52a402452ea | linux-h8ph | 2020-08-28 09:05:57 UTC | 2020-08-28 09:07:02 UTC |                                          |
| SCC_8415767e76ca47978372cc2cc8011f7a | linux-h8ph | 2020-08-28 08:36:21 UTC | 2020-08-28 09:04:08 UTC |                                          |
| SCC_7f14501fc6a14a5ca17c5eaa247d9178 | linux-h8ph | 2020-08-27 15:15:10 UTC | 2020-08-27 15:33:25 UTC |                                          |
| SCC_553b936873664ba1a302590ca17b30b4 | linux-h8ph | 2020-08-27 15:09:32 UTC | 2020-08-27 15:14:32 UTC |                                          |
| SCC_a7e0172feb6a4b949052910ee02b50c2 | linux-h8ph | 2020-08-27 15:08:04 UTC | 2020-08-27 15:09:07 UTC |                                          |
| SCC_b103739340884eee8c415d9f7a62c41e | linux-h8ph | 2020-08-27 15:06:50 UTC | 2020-08-27 15:06:52 UTC |                                          |
| SCC_34dfcff7108448c4afb2ca8d648cda80 | linux-h8ph | 2020-08-27 15:06:32 UTC |                         |                                          |
| SCC_8c0cb836facc4aad9f90591ac2f5322e | linux-h8ph | 2020-08-27 13:51:49 UTC | 2020-08-27 13:52:02 UTC |                                          |
| SCC_6f133f6e8c444c2b8a6175a99ba9ca3f | linux-h8ph | 2020-08-27 13:41:30 UTC | 2020-08-27 13:47:12 UTC |                                          |
| SCC_4163f3ae9a164dbfaea3793e689c6a76 | linux-h8ph | 2020-08-27 13:40:44 UTC |                         |                                          |
| SCC_f006c8ca9abd484ab294008a5b1e7ef6 | linux-h8ph | 2020-08-27 13:36:40 UTC |                         |                                          |
| SCC_a64ff681b97f489ba8b13d1bf3443792 | linux-h8ph | 2020-08-27 13:34:12 UTC | 2020-08-27 13:34:18 UTC |                                          |
| SCC_746e3cd7e30f4d4ca67db686b5765af4 | linux-h8ph | 2020-08-27 13:33:41 UTC | 2020-08-27 13:33:42 UTC |                                          |
| SCC_f11f91b0bdf344c8a9dede45cc558b00 | linux-h8ph | 2020-08-27 13:32:42 UTC | 2020-08-27 13:32:43 UTC |                                          |
| SCC_ee46a677523547688423cc4fcf97c9bb | linux-h8ph | 2020-08-27 13:31:48 UTC | 2020-08-27 13:31:50 UTC |                                          |
| SCC_c2509174f42b43d29298930f38f7f057 | linux-h8ph | 2020-08-27 13:30:15 UTC | 2020-08-27 13:30:17 UTC |                                          |
| SCC_8257dc86dfd34cbfa0e2142ab8b84991 | linux-h8ph | 2020-08-27 13:29:43 UTC | 2020-08-27 13:29:44 UTC |                                          |
| SCC_5ad740e7146e431cb582e8e1115971c0 | linux-h8ph | 2020-08-27 13:15:39 UTC | 2020-08-27 13:15:40 UTC |                                          |
+--------------------------------------+------------+-------------------------+-------------------------+------------------------------------------+
Showing last 20 registrations. Use the '--all' option to see all registered systems.
```

When a system registers to RMT, RMT will create for it a login/password combination and pass it along. The login is the unique identifier that RMT shows to users as it is also the most useful to have.

File: xml/rmt_mirroring.xml (hunk 5 above)
Comment (tmuntaner) [reply xml/rmt_mirroring.xml]: This clean method is still in the documentation as a tip at https://github.com/SUSE/doc-sle/pull/683/files#diff-00a46042cd948e6e2a09f120eb6cb46e8d8a8e35feed59b7c6f0bccc532be1e1R327. The most user-friendly way for a user to clean up RMT's data would be the `rmt-cli repos clean` command and I wanted to show that method first.

File: xml/rmt_mirroring.xml [hunk 8]
```diff
@@ -221,36 +259,76 @@ Jun 22 04:22:34 d31 systemd[1]: Started RMT Mirror timer.</screen>
...
+
+Deleted locally mirrored files from repository 'SLE-Product-SLES15-Updates for sle-15-x86_64'.
+Deleted locally mirrored files from repository 'SLE-Product-SLES15-Pool for sle-15-x86_64'.
+Deleted locally mirrored files from repository 'SLE15-Installer-Updates for sle-15-x86_64'.
+
+Clean finished. An estimated 157 MB were removed.
+   </screen>
+  <tip>
+   <title>Manually Remove Repository Data</title>
+   <para>To delete a disabled repository's data, manually remove its corresponding directory:</para>
```
Comment (tbazant) [inline xml/rmt_mirroring.xml:329]:
```suggestion
   <para>To delete disabled repository data, manually remove its corresponding directory:</para>
```

File: xml/rmt_rmt-cli.xml [hunk 9]
```diff
@@ -258,6 +272,60 @@
...
+     <para>
+      This command forwards registered systems data to SCC.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>rmt-cli systems remove [TARGET]</command></term>
+    <listitem>
+     <para>
+      This command removes a system from RMT by its login.
```
Comment (tbazant) [inline xml/rmt_rmt-cli.xml:325]:
```suggestion
      This command removes a system from RMT as identified by the <guimenu>Login</guimenu> column
      of the output of the <command>rmt-cli systems list</command command.
```

**Review by tbazant — APPROVED** [review]
a few changes and an extended explanation added, now looks good to me

File: xml/rmt_rmt-cli.xml [hunk 10]
```diff
@@ -258,6 +272,61 @@
...
+      This command forwards registered systems data to SCC.
+     </para>
+    </listitem>
+   </varlistentry>
+   <varlistentry>
+    <term><command>rmt-cli systems remove [TARGET]</command></term>
+    <listitem>
+     <para>
+      This command removes a system from RMT as identified by the <guimenu>Login</guimenu> column
+      of the output of the <command>rmt-cli systems list</command command.
```
Comment (tbazant) [inline xml/rmt_rmt-cli.xml:326]:
```suggestion
      This command removes a system from RMT as identified by the
      <guimenu>Login</guimenu> column of the output of the
      <command>rmt-cli systems list</command> command.
```

**Review by tbazant — APPROVED** [review]
one more fix

**Pull request merged**